python/src/energuide/database.py

```diff
@@ -51,13 +53,18 @@ def build_connection_string(coords: DatabaseCoordinates):
     return connection_string
 
 
+@contextmanager  # type: ignore
+def mongo_client(database_coordinates: DatabaseCoordinates) -> typing.Iterable[pymongo.MongoClient]:
+    connection_string = _build_connection_string(database_coordinates)
+    yield pymongo.MongoClient(f'{connection_string}')
+
```

Comment:
Am I missing something, or are you not closing the `MongoClient` connection?

Reply:
Nope, not missing anything. Good call.